src/Contexts/Adaptors/EVMAdaptors/EVMHomeAdaptorProvider.tsx

```diff
@@ -485,6 +499,41 @@ export const EVMHomeAdaptorProvider = ({
     }
   };
 
+  const initFallbackMechanism = async (
+    srcChainId: number,
+    destinationChainId: number,
+    recipient: string,
+    depositNonce: number,
+    decimalAmount: number
+  ): Promise<void> => {
+    const dstChainConfig = getСhainConfig(destinationChainId);
+    const { intervalMs } = getСhainTransferFallback(
+      srcChainId,
+      destinationChainId
+    );
+
+    setInterval(async () => {
+      const cereApi = await createCereApi(dstChainConfig.rpcUrl);
+      const { status } = await getBridgeProposalVotes(
+        cereApi,
+        srcChainId,
+        destinationChainId,
+        recipient,
+        depositNonce,
+        decimalAmount
+      );
+      console.log({ "Proposal votes status": status });
+      switch (status) {
+        case VoteStatus.APPROVED:
+          setTransactionStatus("Transfer Completed");
+          break;
+        case VoteStatus.REJECTED:
+          setTransactionStatus("Transfer Aborted");
+          break;
+      }
+    }, intervalMs);
```

Comment:
The interval from the config is the initial delay. After it, a polling interval should be used. For example, Ethereum -> Cere flow takes ~6 mins. It makes sense to make delay = 1.3*6 mins and after asking the proposal status each 3 * destination network block time seconds.

Reply:
Resolved